src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs

```diff
@@ -511,15 +511,15 @@ public QueryResult<QueryAddressItem> AddressHistory(string address, int offset,
          // This will first perform one db query.
          long total = addressComputedTable.CountSent + addressComputedTable.CountReceived + addressComputedTable.CountStaked + addressComputedTable.CountMined;
 
-         if (offset == 0 || offset > total)
-            offset = (int)total;
-
-         filter = filter.OrderByDescending(s => s.Position);
+         // Filter by the position, in the order of first entry being 1 and then second entry being 2.
+         filter = filter.OrderBy(s => s.Position);
```

Comment:
this is how this look on the explorer
![image](https://user-images.githubusercontent.com/7487930/153760582-3efdf97c-e7c5-4e3c-b11a-cb932b327776.png)

Reply:
This must be handled by logic updated in the explorer. It will retrieve first query with "offset=-1", so it will get the latest page, and the sort order must then be low to high from the API, but reversed in the explorer UI.